app/cmd/rpc/mesh/session.go

```diff
@@ -89,124 +94,497 @@ func (sn DispatchResponse) GetSupportedNodes() []string {
 	return nodes
 }
 
-type AppSessionCache struct {
-	PublicKey       string
-	Chain           string
-	Dispatch        *DispatchResponse
+// NodeSession - contains error/valid information for the node-session relation
+type NodeSession struct {
+	PubKey string
+	// todo: if we can figure out a way to check this, otherwise receive/notify relays until fullNode return evidence sealed.
+	// ^^ that could work with problem, just few "free relays"
 	RemainingRelays int64
+	RelayMeta       *pocketTypes.RelayMeta
+	Validated       bool
+	RetryTimes      int
 	IsValid         bool
 	Error           *SdkErrorResponse
+	Session         *Session
 }
 
-// getAppSession - call ServicerURL to get an application session using retrieve header
-func getAppSession(relay *pocketTypes.Relay, model interface{}) *SdkErrorResponse {
-	servicerNode := getServicerFromPubKey(relay.Proof.ServicerPubKey)
+func (ns *NodeSession) CountRelay() bool {
+	if !ns.Validated {
+		// if this session is not validated yet, will keep been optimistic
+		return true
+	}
+
+	ns.RemainingRelays -= 1
+
+	if ns.RemainingRelays > 0 {
+		return true // still can send relays
+	}
+
+	address, _ := GetAddressFromPubKeyAsString(ns.PubKey)
+
+	logger.Debug(
+		fmt.Sprintf(
+			"servicer=%s exhaust relays for app=%s chain=%s sessionHeight=%d",
+			address,
+			ns.Session.AppPublicKey,
+			ns.Session.Chain,
+			ns.Session.BlockHeight,
+		),
+	)
+
+	ns.IsValid = false
+	ns.Error = NewSdkErrorFromPocketSdkError(pocketTypes.NewOverServiceError(ModuleName))
```

Comment:
should return maximum relays sealed not overservice

Reply:
why? when the evidence can take more relays, should be OverService